### ISSUE TITLE
Nextcloud - don't include /index.php by default in the url

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#1547]
[poisonborz noted in the forums](https://www.turnkeylinux.org/forum/support/mon-20201221-1440/nextcloud-server-will-include-indexphp-default-url) that the Nextcloud/Apache conifg can be tweaked to rewrite the address bar URL that the user sees (and remove the index.php). As suggested/requested, we should add this by default for the v16.1 release IMO.

To repeat the steps required as script:
```
cd /var/www/nextcloud
echo "'htaccess.RewriteBase' => '/'" >> config.php
chown www-data .htaccess
runuser -u www-data php occ maintenance:update:htaccess
chown root .htaccess
systemctl restart apache2
```
FWIW, this requires `mod_rewrite` and `mod_env` (both enabled in the Nextcloud build code by default).

**Comment by vondrt4** [comment]
Please do not do this if it is not supported by original documentation. NextCloud has a lot of plugins and integrations. You need to test:
WebDav mounts, CalDav/CardDav integration
Video calls and a load of other plugins
External sharing links - will they still be generated with index.php in the name? Maybe you need to go into the PHP source code to change it and that would be a lot of work.

**Comment by poisonborz** [comment]
Just to chime in, having `index.php` in the url is "normal" for PHP apps on apache setups not having `mod_rewrite` on, see eg. [Wordpress](https://wordpress.stackexchange.com/questions/105795/remove-index-php-from-permalinks) or [Drupal](https://www.drupal.org/forum/support/post-installation/2016-09-19/how-to-remove-indexphp-from-urls). In no sensible platform/app is the "index.php" part (or the base url) hardcoded, so this shouldn't cause any problems.

Besides .htaccess generation, there also seems to be another way [mentioned in the docs](https://docs.nextcloud.com/server/11.0/admin_manual/configuration_server/config_sample_php_parameters.html), namely setting `'htaccess.IgnoreFrontController' => false` in the config.

**Comment by vondrt4** [comment]
Hmm, the .htaccess generation is automatic and is mentioned in a paragraph just above the config option you mention. Nevertheless, all the things I wrote should be tested before shipping it in the image.

**Comment by poisonborz** [comment]
So the generation is based on the config - I suppose simply adding `'htaccess.RewriteBase' => '/'` before the installation would then also suffice. The docs should then contain a note that if someone would want a `/nextcloud`-like url, he should change the line and re-generate .htaccess.

**Comment by JedMeister** [comment]
Apologies on slow response to this discussion. Thanks @vondrt4 for noting your concerns. The points that you raise are valid. Given that, I think I might just document this for now. Perhaps we could also provide it as a confconsole plugin?

**Comment by OnGle** [comment]
@JedMeister I think we just close this issue at this point

**Comment by JedMeister** [comment]
Yep good call @OnGle - in a perfect world we should probably include it as a doc - but who knows when that will happen, so closing is the go IMO.
